invokeai/frontend/web/src/features/gallery/components/ImageGrid/GalleryImageGrid.tsx

```diff
@@ -66,97 +63,90 @@ const GalleryImageGrid = () => {
+          {
+            categories:
+              galleryView === 'images' ? IMAGE_CATEGORIES : ASSETS_CATEGORIES,
+            is_intermediate: false,
+            offset: 0,
+            limit: INITIAL_IMAGE_LIMIT,
+            ...(selectedBoardId === 'all' ? {} : { board_id: selectedBoardId }),
+          },
+          (draftPosts) => {
+            paginatedData.items.forEach((item) => {
+              draftPosts.items.push(item);
+            });
+          }
+        )
       );
-      return imageList.map((img) => img.image_name);
     }
-  }, [selectedBoardId, galleryView, imagesForBoard, imageNamesAll]);
+    //eslint-disable-next-line
+  }, [paginatedData, dispatch]);
 
   const areMoreAvailable = useMemo(() => {
-    return selectedBoardId === 'all' ? totalAll > imageNamesAll.length : false;
-  }, [selectedBoardId, imageNamesAll.length, totalAll]);
-
-  const isLoading = useMemo(() => {
-    return selectedBoardId === 'all' ? isLoadingAll : isLoadingImagesForBoard;
-  }, [selectedBoardId, isLoadingAll, isLoadingImagesForBoard]);
+    if (imageListResponse?.total) {
+      return imageListResponse?.total > imageList.length;
+    }
+  }, [imageListResponse?.total, imageList.length]);
 
   const handleLoadMoreImages = useCallback(() => {
-    dispatch(
-      receivedPageOfImages({
-        categories:
-          galleryView === 'images' ? IMAGE_CATEGORIES : ASSETS_CATEGORIES,
-        is_intermediate: false,
-        offset: imageNames.length,
-        limit: IMAGE_LIMIT,
-      })
-    );
-  }, [dispatch, imageNames.length, galleryView]);
-
-  const handleEndReached = useMemo(() => {
-    if (areMoreAvailable) {
-      return handleLoadMoreImages;
+    setOffset(imageList.length);
+    setLimit(IMAGE_LIMIT);
+  }, [imageList.length]);
+
+  useEffect(() => {
+    // Set up gallery scroler
+    const { current: root } = rootRef;
+    if (scroller && root) {
+      initialize({
+        target: root,
+        elements: {
+          viewport: scroller,
+        },
+      });
     }
-    return undefined;
-  }, [areMoreAvailable, handleLoadMoreImages]);
-
-  // useEffect(() => {
-  //   if (!didInitialFetch) {
-  //     return;
-  //   }
-  //   // rough, conservative calculation of how many images fit in the gallery
-  //   // TODO: this gets an incorrect value on first load...
-  //   const galleryHeight = rootRef.current?.clientHeight ?? 0;
-  //   const galleryWidth = rootRef.current?.clientHeight ?? 0;
-
-  //   const rows = galleryHeight / galleryImageMinimumWidth;
-  //   const columns = galleryWidth / galleryImageMinimumWidth;
-
-  //   const imagesToLoad = Math.ceil(rows * columns);
-
-  //   setDidInitialFetch(true);
-
-  //   // load up that many images
-  //   dispatch(
-  //     receivedPageOfImages({
-  //       offset: 0,
-  //       limit: 10,
-  //     })
-  //   );
-  // }, [
-  //   didInitialFetch,
-  //   dispatch,
-  //   galleryImageMinimumWidth,
-  //   galleryView,
-  //   selectedBoardId,
-  // ]);
```

Comment:
I removed this just because I wasn't sure what it was doing @psychedelicious

Reply:
This calculates how many images it needs to fetch on first load to fill up the gallery